dessia_common/document_generator.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+""" Write document file. """
+from typing import List, Tuple
+
+import docx
+
+
+class DocxWriter:
+    """ write a docx file. """
+
+    def __init__(self, filename: str):
+        self.filename = filename
+        self.doc = docx.Document()
+        self.headings = None
+
+    def add_headings(self, headings: List[Tuple[str, int]]) -> 'DocxWriter':
+        """ Add a list of headings to the document. """
+        new_doc = self.doc
+        for heading in headings:
+            new_doc.add_heading(heading[0], heading[1])
+        new_writer = DocxWriter(self.filename)
+        new_writer.doc = new_doc
+        new_writer.headings = headings
+        new_writer.add_page_breaks(1)
+        return new_writer
+
+    def add_paragraphs(self, paragraphs: List[str]) -> 'DocxWriter':
+        """ Add a list of paragraphs to the document. """
+        new_doc = self.doc
+        new_headings = self.headings
+        for paragraph in paragraphs:
+            if new_headings:
+                new_doc.add_heading(new_headings[0][0], new_headings[0][1])
+                new_headings = new_headings[1:]
+            new_doc.add_paragraph(paragraph)
+        new_writer = DocxWriter(self.filename)
+        new_writer.doc = new_doc
+        new_writer.headings = new_headings
+        return new_writer
+
+    def add_page_breaks(self, num_page_breaks: int):
+        """ Add an empty page to the document. """
+        for _ in range(num_page_breaks):
+            self.doc.add_page_break()
+
+    def add_table(self, rows: List[List[str]]) -> 'DocxWriter':
+        """ Add table to the document. """
+        new_doc = self.doc
+        table = new_doc.add_table(rows=1, cols=len(rows[0]))
+        for i, cell in enumerate(table.rows[0].cells):
+            cell.text = rows[0][i]
+        for row in rows[1:]:
+            row_cells = table.add_row().cells
+            for i, cell in enumerate(row_cells):
+                cell.text = row[i]
+        new_writer = DocxWriter(self.filename)
+        new_writer.doc = new_doc
+        new_writer.headings = self.headings
+        return new_writer
+
+    def add_header_footer(self, text: str, is_header: bool = True, align: str = 'center') -> 'DocxWriter':
```

Comment:
this should be split into two methods `add_header` and `add_footer`

Reply:
I think it's better that way !

Reply:
For which reason ?

Reply:
because header and  footer it's the same thing just the position (the location) on the document is different

Reply:
Yes, we agree on that point, that why we need to think about the following structure for DocxWriter : 
```

{
    document: Document,
    filename: str,
    headings: ...,
    header: Section (Or Header inheriting from Section, section could be wider),
    footer: Section
}
```

We could also have a "body" attribute, for example, and some other for other stuff.

header and footer would then share the same method that would be called from parent object to generate the right structure (DocxWriter)

Will it be a future dev for you and should I merge this PR now ?

Reply:
Yes, I'm setting this up on my repository,
Yes, you merge it !
tell me if you want me to add the script that I sent you in PM ?